Dangerfile

```diff
@@ -1,119 +1,161 @@
+    private
 
-  ####
+    def error
+      { severity: :error, message: error_message }
+    end
 
-  _Note: We exclude the following files when considering PR size_
+    def error_message
+      <<~EMSG
+        Tooo many lines
+      EMSG
+    end
 
-  ```
-  #{EXCLUSIONS}
+    def warning
+      { severity: :warning, message: warning_message }
+    end
 
-  ```
+    def warning_message
+      <<~EMSG
+        Little too many lines
+      EMSG
+    end
 
-  EXCLUSIONS_TEXT
+    def info
+      { severity: :info, message: 'Good job' }
+    end
 
+    def lines_changed
+      changes.sum(&:total_changes)
+    end
 
-  file_summary = <<~HTML
-    <details><summary>File Summary</summary>
+    def changes
+      `#{files_command}`.split("\n").map do |line|
+        insertions, deletions, file_name = line.split "\t"
+        insertions = insertions.to_i
+        deletions = deletions.to_i
+
+        next if insertions.zero? && deletions.zero?   # Skip unchanged files
+        next if insertions == '-' && deletions == '-' # Skip Binary files
+
+        OpenStruct.new(
+          total_changes: insertions + deletions,
+          insertions: insertions,
+          deletions: deletions,
+          file_name: file_name
+        )
+      end.compact
+    end
 
-    #### Included Files
+    def files_command
+      "git diff #{base_sha}...#{head_sha} --numstat -w --ignore-blank-lines -- . #{exclusions}"
+    end
 
-    - #{included_files.collect { |key, val| "#{key} (+#{val[:insertions]}/-#{val[:deletions]} )" }.join("\n- ")}
-    #{exclusion_text if excluded_files.any?}
-    </details>
-  HTML
+    def exclusions
+      EXCLUSIONS.map { |exclusion| "':!#{exclusion}'" }.join ' '
+    end
 
-  footer = 'Big PRs are difficult to review, often become stale, and cause delays.'
+    def head_sha
+      `git branch --show-current`.chomp
+    end
 
-  if lines_of_code > PR_SIZE[:ABSOLUTE_MAXIMUM]
-    msg = "This PR changes `#{lines_of_code}` LoC (not counting whitespace changes). In order to ensure each PR receives the proper attention it deserves, those exceeding `#{PR_SIZE[:ABSOLUTE_MAXIMUM]}` will not be reviewed, nor will they be allowed to merge. Please break this PR up into smaller ones. If you have reason to believe that this PR should be granted an exception, please see the [Code Review Guidelines FAQ](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/code_review_guidelines.md#faq).\n"
-    fail(msg + file_summary + footer)
-  else
-    msg = "This PR changes `#{lines_of_code}` LoC (not counting whitespace changes). In order to ensure each PR receives the proper attention it deserves, we recommend not exceeding `#{PR_SIZE[:RECOMMENDED_MAXIMUM]}`. Expect some delays getting reviews.\n"
-    warn(msg + file_summary + footer)
+    def base_sha
+      # TODO: decide if it's better to get this info from Danger
```

Comment:
maybe this check doesn't need to run if the target branch isn't master?  I think this is a good concern since sometimes people will make PRs for WiP